theano/gpuarray/pool_grad_grad.c

```diff
@@ -16,9 +17,11 @@ KERNEL void pool_grad_grad_kernel(const ga_size nthreads,
     const ga_size ph = (index / pooled_width) % pooled_height;
     const ga_size c = (index / pooled_width / pooled_height) % channels;
     const ga_size n = (index / pooled_width / pooled_height / channels);
-    const ga_size hstart = ph*stride_h;
+    ga_int hstart = static_cast<ga_int>(ph*stride_h) - static_cast<ga_int>(pad_h);
```

Comment:
I don't think it's a good idea to cast to int here.  If you want a signed type, use ga_ssize.